### DISCUSSION TITLE
Reduced download size of apps

### DISCUSSION THREAD

**Discussion by Universalizer** [bromite/bromite#2432] [General]
Hello, I'm new on github and wanted to discuss a matter, but as on github suggested ; “Discussion section was closed to collaborators only”, so some how I had quoted this previously, you can check this ; https://github.com/bromite/bromite/issues/2412#issue-1423839291,

It is not the issue of any app within itself, but consider it for reducing download size of apps conveniently, by applying compression while zip-align and packing standalone apps installation file.

Thanks to the Developer Team.

**Comment by bat999** [comment]
Mr Google says apk files are zipped files.
Yes we can use an aggressive compressor such as 7z.
But they might not be accepted by very old versions of Android.

**Reply by Universalizer** [reply]
Thanks for your response, for general discussion only.

**Comment by Universalizer** [comment]
(Open-minded is necessary; general discussion only, but few apps names need to be mention ; as a showcase of real live examples, to clearly get well information).

**Example**; one old (again Re-summary) plus four new, total five, all example set for “especially specific to arm32” (Android 32-bit armeabi-v7a only).

> Mr Google says apk files are zipped files. Yes we can use an aggressive compressor such as 7z. But they might not be accepted by very old versions of Android.

Re reply;

What Mr Google had said, probably (may be in) Past Tense, but what google is doing NOW ?, Today in real practice implemented !!! let review herein,

**New Example #2nd, Yesterday** cross-verified that “Brave Beta Browser for Android, arm32 Monoarm Version 1.46.90, apk download size” on Github site is 148 MB”, but Same Ver on Google Play Store ; “new fresh download only (NOT a update) is 60.70 MB”.
Hereby what to understand and extract ?
Anyone, On the web can cross-verify it.

### KIND ATTENTION ;

**Old Example #1st,**

_Kindly Read carefully and understand Specifically Last Paragraph “one real practical highlighting approach” ; https://github.com/bromite/bromite/issues/2412#issue-1423839291,_

**_Oh ! I forgot to add this ; Minimum Required SDK._**

**Example #1st,** Again highlighting and RE-summarizing clearly point wise for better understanding, “ACTUAL size on storage for arm32”, Realistic adopted and implemented approach, Live example here as follows :-

**Bromite Browser :-**
1) Latest Version ; 106.0.5249.163,
2) Min. Required ; SDK Minimum 23 (Marshmallow, Android 6 and above / or later),
3) Actual Download ; 119.17 MB,
4) Actual Installation Occupied ; 133 MB, App Core only (EXCLUDING cache, + other (user) data etc).

**Another CP (Cross-Platform) ;** Firefox Browser for Android (arm32) Real live example here :-
1) Latest Version ; 105.1.0,
2) Min. Required ; SDK Minimum 21 (Lollipop, Android 5 and above / or later),
3) Actual Download ; 73.90 MB (either from F-Droid Repo or Github), or New Version 106.1.0 (from play store, download size is 69.52 MB).
4) Actual Installation Occupied ; 199 MB, App Core only.
5) **NOTE ; Saving approx 63% on Bandwidth.**

Hence above basis example, **Conclusion on “SDK Minimum” ; 23 > 21 , or 21 < 23 (both the case are True).**

In terms of Min. SDK, “Bromite 23 is greater” than, whereas “Firefox, 21 is lesser” , still thousands and millions of people are downloading and installing without any issue.

**_Unprejudiced ; For to get target result, first need is, to try and experiment._**

**Example #3rd, For kind information,** past 24 hour Google Chromium ; org.chromium.chrome, Version 109.0.5403.0 (540300010), requires sdk33/min24. (Min. Android 7, Nougat) soon or later will be “new min. SDK base”.

**Example #4th,** from Play Store ; VLC Player for Android, standalone app size is 34.20 MB, but after Installation 99.75 MB. (Saving 65.71% on Size).

**Example #5th,** There are several apps in google play store with the download size ranges from 20 to 30 MB, but the same apps after installation occupied size becomes 60 to 90 MB.

**May be, now this methodology is effective to flow, in current scenario a general adoptable and already implemented methodology.**

_**Thanks to the Developer Team.**_

**Reply by Universalizer** [reply]
> Unprejudiced ; For to get target result, first need is, to try and experiment.

Sorry if in case, exaggerated by me.

**Comment by Universalizer** [comment]
Ignore this discussion and allied issue. From myside as well as from your side also.
